src/figuresClasses.ts

```diff
@@ -1,19 +1,97 @@
-export interface Figure {
+const ERROR_MESSAGE = 'Value cannot be <= 0!';
 
+function formatNumber(number: number): number {
+  return Math.floor(number * 100) / 100;
 }
 
-export class Triangle {
+enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle'
+}
 
+enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue'
 }
 
-export class Circle {
+export interface Figure {
+  shape: Shape,
+  color: Color,
+  getArea(): number;
+}
+
+export class Triangle implements Figure {
+  public shape: Shape = Shape.Triangle;
+
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
+
+    if (a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('The longest side cannot be >= sum of two others');
+    }
+  }
 
+  getArea(): number {
+    const semiperimeter = (this.a + this.b + this.c) / 2;
+    const area = Math.sqrt(
+      semiperimeter
+        * (semiperimeter - this.a)
+        * (semiperimeter - this.b)
+        * (semiperimeter - this.c),
+    );
+
+    return formatNumber(area);
+  }
 }
 
-export class Rectangle {
+export class Circle implements Figure {
+  public shape: Shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
+  }
 
+  getArea(): number {
+    const area = Math.PI * (this.radius ** 2);
+
+    return formatNumber(area);
+  }
 }
 
-export function getInfo(figure) {
+export class Rectangle implements Figure {
+  public shape: Shape = Shape.Rectangle;
+
+  constructor(
+    public color: Color,
+    public width: number,
+    public height: number,
+  ) {
+    if (width <= 0 || height <= 0) {
+      throw new Error(ERROR_MESSAGE);
+    }
+  }
+
+  getArea(): number {
+    const area = this.width * this.height;
+
+    return formatNumber(area);
+  }
+}
 
+export function getInfo(figure: Figure): string {
+  return `A ${figure.color} ${figure.shape} - ${figure.getArea()}`;
```

Comment:
We can use destructuring, and create variable for this value `figure.getArea()` for better readability